### PR TITLE
fix assertion failure & honor enable_metadata_sync in node operations

### PR DESCRIPTION
**Pull request by onderkalaci**
3 small commits, each self explanatory with comments

### CODE REVIEW

File: src/test/regress/expected/multi_cluster_management.out [hunk 1]
```diff
@@ -1190,6 +1190,52 @@ WHERE logicalrelid = 'test_dist_non_colocated'::regclass GROUP BY nodeport ORDER
...
+(1 row)
+
+	SELECT 1 FROM citus_remove_node('localhost', :worker_2_port);
+ ?column?
+---------------------------------------------------------------------
+        1
+(1 row)
+
+	SELECT 1 FROM citus_add_node('localhost', :worker_1_port);
+NOTICE:  issuing SELECT metadata ->> 'server_id' AS server_id FROM pg_dist_node_metadata
```
Comment (onderkalaci) [inline src/test/regress/expected/multi_cluster_management.out:1220]: these `pg_dist_node_metadata` queries are not metadata sync, so fine to see

Comment (onderkalaci) [inline src/test/regress/expected/multi_cluster_management.out:1220]: these `pg_dist_node_metadata` queries are not metadata sync, so fine to see

File: src/backend/distributed/metadata/node_metadata.c [hunk 2]
```diff
@@ -746,8 +746,6 @@ PgDistTableMetadataSyncCommandList(void)
 	metadataSnapshotCommandList = list_concat(metadataSnapshotCommandList,
 											  colocationGroupSyncCommandList);
 
-	/* As the last step, propagate the pg_dist_object entities */
-	Assert(ShouldPropagate());
```
Comment (agedemenli) [inline src/backend/distributed/metadata/node_metadata.c:750]: Removing the assertion makes sense, but I couldn't see any check with `ShouldPropagate` around here. Are we going to propagate objects to workers even when `citus.enable_ddl_propagation` is set to false?

Comment (onderkalaci) [reply src/backend/distributed/metadata/node_metadata.c:750]: > citus.enable_ddl_propagation

I guess you meant enable_metadata_sync?

> , but I couldn't see any check with ShouldPropagate around here.

I think the assert doesn't make sense, especially inside a function where we create the task list `PgDistTableMetadataSyncCommandList ()`. We can call this function and only print the returning list. Why do we assert `ShouldPropagate` seems like a oversight.

In general, we seem to follow:

```C
if (ThereAreNodesToSync)
{ 
    commands = f();
    .. sync commands
}
```

which makes sense. And, in such structure, `assert` looks redundant (as we  are already inside `if`).

In this case, the caller of `SyncPgDistTableMetadataToNodeList` already filters out the nodes that we should be syncing the metadata. 

One further change I considered was to add `if (!EnableMetadataSync) return` to `SendMetadataCommandListXXX()` function family. But, that is not something I want to do right before the code-freeze.

**Review by agedemenli — APPROVED** [review]
LGTM

**Comment by onderkalaci** [comment]
cherry-picked to release-11.0

**Pull request merged**